projects/laji/src/app/shared-modules/observation-map/observation-map/observation-map.component.ts

```diff
@@ -209,210 +237,272 @@ export class ObservationMapComponent implements OnChanges, OnDestroy {
       }
     }
 
-    const outOfActiveBounds = this.activeZoomThresholdBounds && !this.activeZoomThresholdBounds.contains(e.bounds);
-    const alreadyShowingAll = !this.activeZoomThresholdBounds && this.totalCount < this.showIndividualPointsWhenLessThan;
-    const zoomedInAndNotShowingPoints = this.activeZoomThresholdLevel > curActiveZoomThresholdLevel && !this.showingIndividualPoints;
-    const zoomedOut = this.activeZoomThresholdLevel < curActiveZoomThresholdLevel;
-
-    if (alreadyShowingAll) {
-      return;
-    }
+    const activeBoundsContainViewport = this.activeZoomThresholdBounds && this.activeZoomThresholdBounds.contains(e.bounds);
 
-    if (outOfActiveBounds
-      || zoomedInAndNotShowingPoints
-      || zoomedOut
+    if (
+      !(this.showingIndividualPoints && activeBoundsContainViewport)
+      && (
+           // Zoom in
+           this.activeZoomThresholdLevel > curActiveZoomThresholdLevel
+           // Zoom out
+        || this.activeZoomThresholdLevel < curActiveZoomThresholdLevel
```

Comment:
It's not a biggie, but could be shortened to `this.activeZoomThresholdLevel !== curActiveZoomThresholdLevel`